client/components/Signup.js

```diff
@@ -40,59 +70,92 @@ class Signup extends Component {
-                    {this.state.dirty && (this.state.account.password.length < 6 || this.state.account.password.length > 50) ? (<p>Invalid password</p>) : null}
-                    <button type='submit' disabled={(this.state.account.password.length < 6) || (this.state.account.password.length > 50)}>Create Account!</button>
-                </form>
-                <Google />
+                <Center>
+                    <form onSubmit={this.handleSubmit}>
+                        <TextField
+                            name='username'
+                            floatingLabelText='Username'
+                            type='input'
+                            value={this.state.account.username}
+                            onChange={this.handleChange}
+                            errorStyle={(this.props.message && this.props.message==='Username available') ? {color: 'green'} : {color: 'red'}}
+                            errorText={this.props.message} 
+                            hintText='(Check availability before creating account)'
+                            hintStyle={{ fontSize: '10px' }}
+                        />
+                        <br />
+                        <FlatButton type='button' onClick={this.usernameCheck}>Check Availability</FlatButton>
+                        <br />
+                        <br />
+                        <TextField
+                            name='email'
+                            floatingLabelText='Email'
+                            type='email'
+                            value={this.state.account.email}
+                            onChange={this.handleChange}
+                            errorText={(this.state.dirtyEmail && !(this.validateEmail(this.state.account.email))) ? 'Please enter an email of format: this@example.com' : null}
+                        />
+                        <br />
+                        <br />
+                        <TextField
+                            name='password'
+                            type='password'
+                            floatingLabelText='Password'
+                            hintText='(Must be at least 6 characters long)'
+                            hintStyle={{ fontSize: '10px' }}
+                            value={this.state.account.password}
+                            onChange={this.handleChange}
+                            errorText={this.state.dirtyPassword && (this.state.account.password.length < 6 || this.state.account.password.length > 50) ? 'Invalid password' : null}
+                        />
+                        <br />
+                        <Center>
+                            <div>
+                                 <FlatButton type='submit' disabled={((this.state.account.password.length < 6) || (this.state.account.password.length > 50) || (!this.validateEmail(this.state.account.email)) || (this.props.message==='Username not available') || (this.props.message===''))}>Create Account!</FlatButton> 
```

Comment:
If it is disabled because user has not verified the user name, you need to somehow signal that to the user.

Reply:
There's already a hint for the user to check availability first.